url026791d51b_dhleong_SeriesTracker_v6_tgz-pJ8-stv6_SeriesTrackerJ8/src/stv6/http/pools/ClientPool.java

```diff
@@ -0,0 +1,113 @@
+package stv6.http.pools;
+
+import java.util.LinkedList;
+import java.util.concurrent.Semaphore;
+
+import stv6.http.Client;
+
+/**
+ * A client pool should allow for multiple, threaded
+ * 	clients, without the risk of over-running Java's
+ * 	thread limit. Because, really, we probably don't
+ * 	need a separate thread for each client
+ * 
+ * For now, we'll be lazy
+ * 
+ * @author dhleong
+ *
+ */
+public class ClientPool implements ClientList {
+    
+    /**
+     * If we don't get data within 15 seconds, timeout
+     */
+	public static final long TIMEOUT_DELAY = 15000;
+
+    private final class ClientThread implements Runnable {
+
+	    private final boolean mRunning = true;
+		public String name;
+
+        @Override
+        public void run() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 13. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.